### PR TITLE
docs: sd-drawer docs  [skip chromatic] 

### PR DESCRIPTION
**Pull request by smfonseca**
## Definition of Reviewable:

- [ ] Documentation is created/updated
- [ ] Migration Guide is created/updated
- [ ] E2E tests (features, a11y, bug fixes) are created/updated
- [ ] Stories (features, a11y) are created/updated
- [x] relevant tickets are linked

### CODE REVIEW

**Comment by yoezlem** [comment]
Not sure how to handle it but in design we are showing it a bit different:

Figma:
image

Storybook:
image

@coraliefeil @MartaPintoTeixeira @smfonseca we should base it on one of the screens.

**Review by yoezlem — CHANGES_REQUESTED** [review]
Whats with the padding story? don't we need that in storybook? If so, we should mark it in Figma with a star @coraliefeil @MartaPintoTeixeira @smfonseca 

image

**Comment by yoezlem** [comment]
@smfonseca I miss the size from the Figma file. Please align with the design.

File: packages/components/src/components/drawer/drawer.test.stories.ts [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+export default {
+  title: 'Components/sd-drawer/Screenshot Tests',
+  tags: ['!autodocs'],
+  component: 'sd-drawer',
+
+  args: overrideArgs([
+    {
+      type: 'slot',
+      name: 'default',
+      value: `<div class="slot slot--border slot--text h-full">Main slot</div>`
```
Comment (yoezlem) [inline packages/components/src/components/drawer/drawer.test.stories.ts:21]:
```suggestion
      value: `<div class="slot slot--border slot--text h-full">Default slot</div>`
```

File: packages/components/src/components/drawer/drawer.test.stories.ts [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+          axis: {
+            x: {
+              type: 'slot',
+              name: slot,
+              title: 'slot=...',
+              values: [
+                {
+                  value:
+                    slot === 'default'
+                      ? `<div class="slot slot--border slot--background slot--text h-full">Main slot</div>`
```
Comment (yoezlem) [inline packages/components/src/components/drawer/drawer.test.stories.ts:287]:
```suggestion
                      ? `<div class="slot slot--border slot--background slot--text h-full">Default slot</div>`
```

File: packages/components/src/components/drawer/drawer.stories.ts [hunk 3]
```diff
@@ -43,371 +44,74 @@ export default {
...
-      if (header) {
-        await userEvent.type(header, '{space}', { pointerEventsCheck: 0 });
-      }
-    }
-  }
+        <sd-button slot="footer">Lorem</sd-button>
+      </sd-drawer>
+    </div>
+  `
 };
```
Comment (yoezlem) [inline packages/components/src/components/drawer/drawer.stories.ts:117]: Please do it like in the figma file:
image

Comment (smfonseca) [reply packages/components/src/components/drawer/drawer.stories.ts]: Updated

**Review by yoezlem — APPROVED** [review]

File: packages/components/src/templates/drawer-with-navigation.stories.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  name: 'Drawer with Navigation item Template',
+  render: () => html`
+    <style>
+      sd-navigation-item::part(content) {
+        display: flex;
+        align-items: center;
+      }
+    </style>
+
+    <sd-drawer open placement="start">
```
Comment (coraliefeil) [inline packages/components/src/templates/drawer-with-navigation.stories.ts:36]: "end"

File: packages/components/src/templates/drawer-with-navigation.stories.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            background-color: #ecf0f9;
+            min-height: 750px;
+          }
+        </style>
+        ${story()}`
+  ] as unknown
+};
+
+export const Default = {
+  name: 'Drawer with Navigation item Template',
```
Comment (coraliefeil) [inline packages/components/src/templates/drawer-with-navigation.stories.ts:27]: I added [more templates](https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=3000-10344&node-type=SECTION&t=vj79pLa8rdsBYND3-11) which we can add at a later point.

For now we can keep the template we already have.

Pls adjust it according to design:

image


Placement, blanket color, add a border so one can understand the dimension of the viewport, adjust inner padding ...

File: packages/components/src/components/drawer/drawer.stories.ts [hunk 6]
```diff
@@ -36,378 +39,87 @@ export default {
...
-            value: ''
-          }
-        ],
-        args
-      })}
-    </div>`;
-  }
-};
+export const Open = {
+  name: 'Open',
```
Comment (coraliefeil) [inline packages/components/src/components/drawer/drawer.stories.ts:74]: We said we would like to show this for open:

image

**Comment by coraliefeil** [comment]
One last thing ... see comment

**Review by yoezlem — APPROVED** [review]

File: packages/components/src/components/drawer/drawer.stories.ts (hunk 6 above)
Comment (smfonseca) [reply packages/components/src/components/drawer/drawer.stories.ts:74]: Updated

**Comment by coraliefeil** [comment]
Approved. THX!

**Pull request merged**